pom.xml

```diff
@@ -94,10 +94,9 @@
   </distributionManagement>
 
   <properties>
-    <java.version>6</java.version>
+    <java.version>8</java.version>
```

Comment:
If I'm right Spring 5's base is JDK 8, but Spring 4's is 6 or 7, right? Has this java base version been upped to 8 because of Jakarta EE's requirements? If so, wouldn't this an issue for keeping Spring 4 support?

I think Spring 3 support can perfectly be removed in Thymeleaf 3.1, but I am a bit doubtful about Spring 4… not that I would be against that at all, to be honest –I can see the advantages–, but nevertheless I would like to know the possibilities...

Reply:
You are correct that this was updated because of Jakarta EE's requirements (see commit log). 

You are also right that Spring 5 was the first to require JDK 8, so if we kept the arrangement as is we'd need to remove support for older Spring versions. Honestly, I think that is the responsible thing to do at this point since [Spring 4 reached EOL December 31, 2020](https://spring.io/blog/2019/12/03/spring-framework-maintenance-roadmap-in-2020-including-4-3-eol).

If we wanted to preserve the old version, then we could place the Jakarta code in a separate module. I have mixed emotions on continuing to use JDK 6. On one hand updating to JDK 8 is a breaking change. On the other hand, JDK 6 is no longer supported.